### PR TITLE
Betamocks integration

### PR DESCRIPTION
**Pull request by kreek**
_Update on 10-31_ :jack_o_lantern:
Bumping this PR @kreek @saneshark . Notes:
- All GIBS & MVI mock stuff (service_factory, mock data) removed from `vets-api` - now soley dependent on `betamocks` + `vets-api-mockdata`
- New setting: `Settings.emis.mock`
- `betamocks` can be disabled globally with the `Settings.betamocks.enabled: false`
- Mock can be enabled/disabled per service:
  - `Settings.mvi.mock`
  - `Settings.emis.mock`
  - `Settings.evss.mock_letters`
  - `Settings.evss.mock_gi_bill_status`
  - `Settings.evss.mock_pciu_address`

@lihanli I noticed you tried to consolidate our base service for EVSS.  I had to add some config to various parts of code and having a unified base service class would help.

### CODE REVIEW

**Comment by saneshark** [comment]
@kreek I'm currently using VCR to do integration spec recording, you can see the code in initializers/integration_recorder.rb and then I'm using those generated cassettes (for both incoming vets-website requests and outgoing external requests) to play back interactions for E2E testing of certain things such as our login flow. To facilitate I've used Timecop to freeze time according to what's in the spec, but one problem with this approach has been that VCR records the time using httpdate and strips the milliseconds. Does your gem record the millisecond accuracy? If so, I would probably refactor my stuff to use that instead.

**Comment by kreek** [comment]
@saneshark do you mean VCR's `recorded_at` in the yaml file or the date in the header? If it's `recorded_at` Betamocks doesn't have a similar field but it wouldn't be difficult to add and could include milliseconds.

**Comment by saneshark** [comment]
That sounds awesome. I'll DM you. I'm not crazy about that stupid initializer. Furthermore I'm less crazy about having to override SecureRandom the way I'm doing. I would have used ruby 2.3 `refinements` https://ruby-doc.org/core-2.3.0/doc/syntax/refinements_rdoc.html but they don't support modules, only classes. In any case, I override the base method in SecureRandom and pass it a Kernel.srand based on the frozen Time.now.to_i, additionally on current_user where we record last_signed_in date and things of that nature, the JSON includes milliseconds, so when I freeze time based on the yaml the JSON doesn't match... These are minor issues relatively speaking, but real opportunity to think through and address them with a gem like yours. I'm currently using VCR's filtering which is nice for hiding urls and tokens in the generated yaml. Also another feature I'm not sure if you've implemented, but I could totally handle that in my middleware too I think.

**Comment by omgitsbillryan** [comment]
Two files comically under our 90% coverage threshold @ 89.47% & 89.90%.

@kreek do these statuses even get used anymore? (coverage report screenshot)
image

**Comment by kreek** [comment]
@omgitsbillryan I think we can pull that out now that we have more robust error reporting elsewhere

**Comment by omgitsbillryan** [comment]
Decided not to remove it since there's FE code that depends on it, and many other response types subclass from it.  Seems to be a point of tech debt, so I logged a separate issue:

https://github.com/department-of-veterans-affairs/vets.gov-team/issues/5727

File: config/betamocks/services_config.yml [hunk 1]
```diff
@@ -58,11 +58,11 @@
   :base_uri: <%= URI(Settings.mvi.url).host %>
   :endpoints:
   - :method: :post
-    :path: "/psim_webservice/IdMWebService"
+    :path: <%= URI(Settings.mvi.url).path %>
```
Comment (omgitsbillryan) [inline config/betamocks/services_config.yml]: 👍 makes a lot of sense

File: config/betamocks/services_config.yml [hunk 2]
```diff
@@ -58,11 +58,11 @@
...
     :file_path: "mvi/profile"
     :cache_multiple_responses:
       :uid_location: body
-      :uid_locator: 'root="2.16.840.1.113883.4.1" extension="(\d{9})"'
+      :uid_locator: '(?:root="2.16.840.1.113883.4.1" )?extension="(\d{9})"(?: root="2.16.840.1.113883.4.1")?'
```
Comment (omgitsbillryan) [inline config/betamocks/services_config.yml]: AFAI knew, this already worked for local dev environments.  I assume you're making this change for `dev` or `staging`.  Can you confirm that this still works for local dev?

Comment (patrickvinograd) [reply config/betamocks/services_config.yml]: Yes, we are seeing that Ox can generate the MVI request with these attributes in either order. This regex accepts both variants.

**Pull request merged**